{Doc} Track 2 Migration Guidance (#16484)
Azure/azure-cli: doc/track_2_migration_guidance.md

### Long running operation function name change

> Long running operations have changed their function names in Track 2 SDK. A `begin_` prefix is added. E.g. `create_or_update` becomes `begin_create_or_update`. `delete` becomes `begin_delete`. It is a naming convention in Track 2 SDK to indicate that an operation is a long running operation. Test cases can uncover most instances. But if a command has no test, it will be missed. A rigorous method is going through all methods to see whether they are long running operations.

Reviewer: I prefer the word "reveal", because "uncover" is usually for something secret:

https://www.collinsdictionary.com/us/dictionary/english/uncover

> If you **uncover** something, especially something that has been kept secret, you discover or find out about it.

```suggestion
Long running operations have changed their function names in Track 2 SDK. A `begin_` prefix is added. E.g. `create_or_update` becomes `begin_create_or_update`. `delete` becomes `begin_delete`. It is a naming convention in Track 2 SDK to indicate that an operation is a long running operation. Test cases can reveal most instances, but if a command has no test, it may be missed. A reliable approach is going through all methods to see whether they are long running operations.
```

Reviewer: "e.g." is usually place at the end of a sentence.

https://www.collinsdictionary.com/us/dictionary/english/eg

> We need helpers of all types, e.g., geologists and teachers.

```suggestion
Long running operations have changed their function names in Track 2 SDK. A `begin_` prefix is added. For example, `create_or_update` becomes `begin_create_or_update`. `delete` becomes `begin_delete`. It is a naming convention in Track 2 SDK to indicate that an operation is a long running operation. Test cases can uncover most instances. But if a command has no test, it will be missed. A rigorous method is going through all methods to see whether they are long running operations.
```

> Long running operations have changed their function names in Track 2 SDK. A `begin_` prefix is added. E.g. `create_or_update` becomes `begin_create_or_update`. `delete` becomes `begin_delete`. It is a naming convention in Track 2 SDK to indicate that an operation is a long running operation. Test cases can reveal most instances, but if a command has no test, it may be missed. A reliable approach is going through all methods to see whether they are long running operations.

Author: ```suggestion
Long running operations have changed their function names in Track 2 SDK. A `begin_` prefix is added. For example, `create_or_update` becomes `begin_create_or_update`. `delete` becomes `begin_delete`. It is a naming convention in Track 2 SDK to indicate that an operation is a long running operation. Test cases can reveal most instances, but if a command has no test, it may be missed. A reliable approach is going through all methods to see whether they are long running operations.
```
